simuleval/evaluator/instance.py

```diff
@@ -338,25 +338,31 @@ def prediction(self):
 
     def summarize(self):
         samples = []
-        # start from the first segment offset
-        start = prev_end = self.delays[0]
         self.intervals = []
         self.silences = []
 
-        for i, delay in enumerate(self.delays):
-            start = max(prev_end, delay)
+        if len(self.prediction_list) > 0:
+            # start from the first segment offset
+            start = prev_end = prediction_offset = self.delays[0]
 
-            if start > prev_end:
-                # Wait source speech, add discontinuity with silence
-                samples += [0.0] * int(
-                    self.target_sample_rate * (start - prev_end) / 1000
-                )
-                self.silences.append(start - prev_end)
+            for i, delay in enumerate(self.delays):
+                start = max(prev_end, delay)
 
-            samples += self.prediction_list[i]
-            duration = self.durations[i]
-            prev_end = start + duration
-            self.intervals.append([start, duration])
+                if start > prev_end:
+                    # Wait source speech, add discontinuity with silence
+                    samples += [0.0] * int(
+                        self.target_sample_rate * (start - prev_end) / 1000
+                    )
+                    self.silences.append(start - prev_end)
+
+                samples += self.prediction_list[i]
+                duration = self.durations[i]
+                prev_end = start + duration
+                self.intervals.append([start, duration])
+        else:
+            # For empty prediction
+            prediction_offset = self.source_length
+            samples = [0.0] * 1000
```

Comment:
IIRC, we make the sample to an all-zero list of length of 1000 (1000/fs second). What's the reason for this? Can you just skip the `soundfile.write` if the output is empty?

Reply:
Sounds cool! updated